packages/frontend/src/hooks/useCustomVisualizationConfig.ts

```diff
@@ -0,0 +1,74 @@
+import { ApiQueryResults, CustomVis, ResultRow } from '@lightdash/common';
+import { useEffect, useMemo, useState } from 'react';
+
+const convertRowsToSeries = (rows: ResultRow[]) => {
+    return rows.map((row) => {
+        return Object.fromEntries(
+            Object.entries(row).map(([key, rowValue]) => [
+                key,
+                rowValue.value.raw,
+            ]),
+        );
+    });
+};
+
+export interface CustomVisualizationConfigAndData {
+    validConfig: CustomVis;
+    visSpec?: string;
+    setVisSpec: (spec: string) => void;
+    // TODO: do we need to type this better?
+    series: {
+        [k: string]: unknown;
+    }[];
+    fields?: string[];
+}
+
+const useCustomVisualizationConfig = (
+    chartConfig: CustomVis | undefined,
+    resultsData: ApiQueryResults | undefined,
+): CustomVisualizationConfigAndData => {
+    const [visSpec, setVisSpec] = useState<string | undefined>();
+    const [visSpecObject, setVisSpecObject] = useState();
+
+    // Set initial value
+    useEffect(() => {
+        try {
+            if (chartConfig?.spec && !visSpec) {
+                setVisSpec(JSON.stringify(chartConfig?.spec, null, 2));
+            }
+        } catch (e) {
+            //TODO: handle error
+        }
+    }, [chartConfig?.spec, visSpec]);
+
+    // Update object when spec changes
+    useEffect(() => {
+        try {
+            if (visSpec) {
+                setVisSpecObject(JSON.parse(visSpec));
+            }
+        } catch (e) {
+            //TODO: handle error
+        }
+    }, [visSpec]);
```

Comment:
I think there's a bug somewhere where this happens. happened when I was testing locally:

https://github.com/lightdash/lightdash/assets/962095/b2e44bc5-8e37-4105-abc8-acf139c553a3

Reply:
Good find. I will fix this in a follow up if you dont mind. In some ways its a feature -- this spec cannot start with any character except '{' anyway. But I will fix.